packages/core/apps/core-apps-server-internal/src/core_app.test.ts

```diff
@@ -25,16 +26,17 @@ const emptyPlugins = (): UiPlugins => ({
 
 describe('CoreApp', () => {
   let coreContext: ReturnType<typeof mockCoreContext.create>;
-  let coreApp: CoreApp;
-  let internalCorePreboot: ReturnType<typeof coreMock.createInternalPreboot>;
+  let coreApp: CoreAppsService;
+  let internalCorePreboot: ReturnType<typeof coreInternalLifecycleMock.createInternalPreboot>;
```

Comment:
As of https://github.com/elastic/kibana/issues/141891, core's internal lifecycles are exposed in a package, so we can use those directly.